pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -718,59 +750,84 @@ func TestInsufficientCapacityNodeDaemonDoesNotLaunchPod(t *testing.T) {
 
 // DaemonSets should not unschedule a daemonset pod from a node with insufficient free resource
 func TestInsufficientCapacityNodeDaemonDoesNotUnscheduleRunningPod(t *testing.T) {
-	for _, strategy := range updateStrategies() {
-		podSpec := resourcePodSpec("too-much-mem", "75M", "75m")
-		podSpec.NodeName = "too-much-mem"
+	for _, f := range []bool{true, false} {
+		setFeatureGate(t, features.ScheduleDaemonSetPods, f)
+		for _, strategy := range updateStrategies() {
+			podSpec := resourcePodSpec("too-much-mem", "75M", "75m")
+			podSpec.NodeName = "too-much-mem"
+			ds := newDaemonSet("foo")
+			ds.Spec.UpdateStrategy = *strategy
+			ds.Spec.Template.Spec = podSpec
+			manager, podControl, _, err := newTestController(ds)
+			if err != nil {
+				t.Fatalf("error creating DaemonSets controller: %v", err)
+			}
+			node := newNode("too-much-mem", nil)
+			node.Status.Allocatable = allocatableResources("100M", "200m")
+			manager.nodeStore.Add(node)
+			manager.podStore.Add(&v1.Pod{
+				Spec: podSpec,
+			})
+			manager.dsStore.Add(ds)
+			switch strategy.Type {
+			case apps.OnDeleteDaemonSetStrategyType:
+				if !utilfeature.DefaultFeatureGate.Enabled(features.ScheduleDaemonSetPods) {
+					syncAndValidateDaemonSets(t, manager, ds, podControl, 0, 0, 2)
+				} else {
+					syncAndValidateDaemonSets(t, manager, ds, podControl, 1, 0, 0)
```

Comment:
Depending on DS Scheduling enabled or not, the number of events would vary, create in this case is varying because previously, we stopped even creating pod after delete when there were no sufficient resources. Now, as you suggested we are creating pod irrespective of resource crunch, that is why we need this branch. Without this, we have to refactor the `syncAndValidateDaemonSets` fn